rubrics/impl/src/main/java/org/sakaiproject/rubrics/impl/RubricsServiceImpl.java

```diff
@@ -366,6 +366,29 @@ public Optional<CriterionTransferBean> createDefaultCriterion(String siteId, Lon
         });
     }
 
+    public Optional<CriterionTransferBean> createDefaultEmptyCriterion(String siteId, Long rubricId) {
+
+        String currentUserId = sessionManager.getCurrentSessionUserId();
+
+        if (StringUtils.isBlank(currentUserId) || !isEditor(siteId)) {
+            throw new SecurityException("You must be a rubrics editor to create/edit criteria");
+        }
+
+        return rubricRepository.findById(rubricId).map(rubric -> {
```

Comment:
I'd get the rubric with an orElseThrow and an illegalargumentexception

Rubric rubric = rubricRepository.findById(rubricId).orElseThrow(() -> new IllegalArgumentException("No rubric for id " + rubricId));

Reply:
I know the map stuff's really cool, but it makes sense to me to add some handling for a bad rubricId.

Reply:
But, it's up to you